src/AsyncData.ts

```diff
@@ -269,14 +269,14 @@ class __AsyncData<A> {
     return mapper((this as Done<A>).value);
   }
 
-  match<B>(
+  match<B1, B2 = B1, B3 = B1>(
```

Comment:
do the default assignments do anything there?

Reply:
Yes, they allow you to enforce a type without enforcing all of them.

Ex, you want to force the result to be `Toto`:
```ts
Result.match<Toto>()` -> will enforce `Toto` return type for both `Ok` and `Error` // if we don't add default assignement, you cannot only enforce `Toto`, have to explicitely enforce `<Toto, Toto>`

Result.match<Toto, Tata>()` -> will enforce `Toto` return type for `Ok`, `Tata` on `Error`
```